pkg/util/config_util.go

```diff
@@ -0,0 +1,130 @@
+package util
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// CreateIfNotExists creates the directory and the file if it doesn't exist
+func CreateIfNotExists(configFile string) error {
+	_, err := os.Stat(filepath.Dir(configFile))
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(filepath.Dir(configFile), 0755)
+		if err != nil {
+			return errors.Wrap(err, "unable to create directory")
+		}
+	}
+	// Check whether config file is present or not
+	_, err = os.Stat(configFile)
+	if os.IsNotExist(err) {
+		file, err := os.Create(configFile)
+		if err != nil {
+			return errors.Wrap(err, "unable to create config file")
+		}
+		defer file.Close()
+	}
+
+	return nil
+}
+
+// GetFromFile unmarshals a struct from a odo config file
+func GetFromFile(c interface{}, filename string) error {
+	configData, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return errors.Wrapf(err, "unable to read file %v", filename)
+	}
+
+	err = yaml.Unmarshal(configData, c)
```

Comment:
Looks like there is an asymmetry between Get and Write that I don't like. Doesn't this assume that `c` is a pointer here? While, in `WriteToFile`, `c` is assumed to be a `struct` since we take its address? If we keep the asymmetry shouldn't we check that `c` is indeed of the proper type (pointer or not) at least?

Reply:
resolved, now both assume pointers